runtime/src/main/kotlin/kotlin/Numbers.kt

```diff
@@ -100,4 +100,180 @@ public actual inline fun Float.Companion.fromBits(bits: Int): Float = kotlin.fro
 
 @PublishedApi
 @TypedIntrinsic(IntrinsicType.REINTERPRET)
-external internal fun fromBits(bits: Int): Float
+internal external fun fromBits(bits: Int): Float
+
+
+/**
+ * Counts the number of set bits in the binary representation of this [Int] number.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+@SymbolName("Kotlin_Int_countOneBits")
+public actual external fun Int.countOneBits(): Int
+
+/**
+ * Counts the number of consecutive most significant bits that are zero in the binary representation of [Int] [value].
+ * Returns undefined result for zero [value].
+ */
+@SymbolName("Kotlin_Int_countLeadingZeroBits")
+private external fun countLeadingZeroBits(value: Int): Int
+
+/**
+ * Counts the number of consecutive most significant bits that are zero in the binary representation of this [Int] number.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+public actual fun Int.countLeadingZeroBits(): Int =
+        if (this == 0) Int.SIZE_BITS else countLeadingZeroBits(this)
+
+/**
+ * Counts the number of consecutive least significant bits that are zero in the binary representation of [Int] [value].
+ * Returns undefined result for zero [value].
+ */
+@SymbolName("Kotlin_Int_countTrailingZeroBits")
+private external fun countTrailingZeroBits(value: Int): Int
+
+/**
+ * Counts the number of consecutive least significant bits that are zero in the binary representation of this [Int] number.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+public actual fun Int.countTrailingZeroBits(): Int =
+        if (this == 0) Int.SIZE_BITS else countTrailingZeroBits(this)
+
+/**
+ * Returns a number having a single bit set in the position of the most significant set bit of this [Int] number,
+ * or zero, if this number is zero.
+ */
+@SinceKotlin("1.3")
+@ExperimentalStdlibApi
+public actual fun Int.takeHighestOneBit(): Int =
+        if (this == 0) 0 else 1.shl(Int.SIZE_BITS - 1 - countLeadingZeroBits(this))
```

Comment:
I'd use 32 here, as access to companion object looks like an overkill here.

Reply:
As we do not have constant propagation, and those functions intended to be on perf sensitive paths.

Reply:
Ok, I'll inline these constants manually, but leave a TODO to replace them back when constant propagation is ready. Btw, is there an issue about it to follow?